airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/convert/AirbyteValueToStatement.kt

```diff
@@ -88,16 +90,50 @@ class AirbyteValueToStatement {
             if (sqlType == Types.VARCHAR || sqlType == Types.LONGVARCHAR) {
                 setString(idx, value.value)
             } else {
-                throw IllegalArgumentException("$value isn't a $type")
+                // TODO: this is a fallback because Values aren't fully typed.
```

Comment:
@jdpgrailsdev, this is the part I changed to avoid rewriting the whole statement prep right now.
Ideally, in a following iteration of the CDK, there should be an option to fully type only top level so we shouldn't need this hack. That version of the CDK may also involve rewriting a decent part of the type handling so I am leaving refactoring for when that happens.